### PR TITLE
[stable/insights-agent] Install-reporter securityContext template and default values

### PR DESCRIPTION
**Pull request by Azahorscak**
**Why This PR?**
This makes the securityContext of the install reporter section in the insights-agent chart configurable. 
Also includes relatively strict security best practices in the default values. 

Fixes #

**Changes**
Changes proposed in this pull request:

* Allows configuration of the securityContext for the install-reporter
* uses good defaults

**Checklist:**

* [x] I have included the name of the chart in the title of this PR in square brackets i.e. `[stable/goldilocks]`.
* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

### CODE REVIEW

**Review by rbren — APPROVED** [review]
Thanks Adam!

File: stable/insights-agent/values.yaml [hunk 1]
```diff
@@ -71,6 +71,15 @@ installReporter:
       cpu: 100m
       memory: 128Mi
   additionalAnnotations: {}
+  securityContext:
+    allowPrivilegeEscalation: false
+    readOnlyRootFilesystem: true
+    runAsNonRoot: true
+    runAsUser: 1000
```
Comment (rbren) [inline stable/insights-agent/values.yaml:79]: are we missing `privileged: false` here?

Comment (Azahorscak) [reply stable/insights-agent/values.yaml:79]: not anymore, Thanks!

File: scripts/e2e-test.sh [hunk 2]
```diff
@@ -31,6 +31,11 @@ teardown () {
     kind delete cluster --name="$TEST_CLUSTER_NAME"
 }
 
+# delete_jobs () {
+#     printf "Deleting all deployed jobs...\n"
+#     kubectl delete jobs "kubectl get jobs -o custom-columns=:.metadata.name"
+# }
+
```
Comment (sudermanjr) [inline scripts/e2e-test.sh]:
```suggestion

```

File: scripts/e2e-test.sh [hunk 3]
```diff
@@ -52,6 +57,7 @@ elif [ "$OPERATION" = "teardown" ] ; then
 elif [ "$OPERATION" = "test" ]; then
     printf "Running tests.\n"
     pre_test_script
+    # delete_jobs
```
Comment (sudermanjr) [inline scripts/e2e-test.sh]:
```suggestion
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]

**Review by sudermanjr — APPROVED** [review]

**Pull request merged**